### PR TITLE
Replaces Camera bug with Camera tablet app

### LINKED ISSUE #39975: Camera bugs do not work
Trying to use a camera bug opens up a blank window instead of a list of cameras or camera view as it should. The bug can still be used on security cameras and will display "Camera Bugged" when it is used, but interacting with the camera bug item still only shows a blank window.

### LINKED ISSUE #53820: Camera bug camera view is affected by vignette of being in a locker
Issue reported from  Round ID: [146533](https://scrubby.melonmesa.com/round/146533) (/tg/Station Sybil [ENGLISH] [US-WEST] [100% LAG FREE])
Reporting client version: 513.1532

Pretty sure the screen is INSIDE the locker on the device and so shouldn't be affected.

### LINKED ISSUE #74839: Civilian console can't display correctly SecurEye camera software
Reporting client version: 514.1589

## Round ID:
[204027](https://scrubby.melonmesa.com/round/204027)

## Reproduction:
1. Open any civilan console, and try to install SecurEye on it.
2. See the interface with cam selection working, but instead of image - black screens.
3. Dance!

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Replaces the Traitor's Camera Bug item with an application that allows you to view camera apps.
The difference between this app and the Security one is that it does not have an access requirement, does not make noise (it didn't previously due to a bug, fixed in this PR), and can be installed on PDAs.
This can also be installed from syndienet, which means nukies now have a way to see the station cameras while off the ship.

Adds Syndicate app disks, which are made of Gold. This is only used by the Camera app as of right now.

I also fixed some issues along the way;
- Camera tablet app now properly shows cameras
- It now properly makes the noises it is supposed to
- It clears the viewers properly on the ui being closed or the app being exited.

Syndicate app disks ddelete their apps upon being transferred over (like maintenance disks), trying to remain consistent with previous behavior.

I also made this for a [personal project I am currently working on](https://hackmd.io/XLt5MoRvRxuhFbwtk4VAUA). I could just make this UI be TGUI, but I thought it could be worth trying to turn it into a better item first.

Tracking people is much simpler now- You choose anyone from a list of people found on cameras, and it will try to find a camera that sees them. It follows them until you close the app or switch camera.
https://github.com/tgstation/tgstation/assets/53777086/1536ebb9-0c4f-45bb-b593-f98791ea6d23

## Why It's Good For The Game

The Camera Bug is one of the worst traitor items as of current. You can remotely shut off a single camera, something that can be done with basic tools you are likely carrying around anyways, and uses an HTML clunky UI to flip through cameras.
This new Traitor item makes it much easier to use, since the camera consoles are something you are likely already used to.
It also means emagging a tablet is slightly more useful, and golden data disks are pretty cool.

Unfortunately this means that they no longer grant illegal tech, because otherwise you can simply clone the app infinitely for personal use, and deconstruct the disk itself for tech. This can be grinded into gold, but I think if we want an illegal tech item, it'll have to find a new host.

Closes https://github.com/tgstation/tgstation/issues/74839
Closes https://github.com/tgstation/tgstation/issues/39975
Closes https://github.com/tgstation/tgstation/issues/53820

## Changelog

:cl:
balance: The Traitor's Camera bug is now a tablet app that works like a silent camera console with an extra ability to track people through their nearby cameras.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
closes https://github.com/tgstation/tgstation/issues/53820 closes https://github.com/tgstation/tgstation/issues/39975 by virtue of new behavior

File: code/_globalvars/bitfields.dm [hunk 1]
```diff
@@ -282,18 +282,18 @@ DEFINE_BITFIELD(movement_type, list(
 ))
 
 DEFINE_BITFIELD(obj_flags, list(
-	"BLOCK_Z_IN_DOWN" = BLOCK_Z_IN_DOWN,
-	"BLOCK_Z_IN_UP" = BLOCK_Z_IN_UP,
-	"BLOCK_Z_OUT_DOWN" = BLOCK_Z_OUT_DOWN,
-	"BLOCK_Z_OUT_UP" = BLOCK_Z_OUT_UP,
+	"IN_USE" = IN_USE,
```
Comment (JohnFulpWillard) [inline code/_globalvars/bitfields.dm:285]: This is just reorganized in order in which they are defined

**Comment by Helg2** [comment]
Shouldn't you replace bugs with disks on the maps?

**Comment by JohnFulpWillard** [comment]
> Shouldn't you replace bugs with disks on the maps?

I can, didn't really think about it much

**Comment by newfren** [comment]
camera bugs are currently a common source of illegal tech in space, will these disks also have illegal tech and replace the camera bugs?

**Comment by JohnFulpWillard** [comment]
> camera bugs are currently a common source of illegal tech in space, will these disks also have illegal tech and replace the camera bugs?

currently yes but perhaps that is not a good idea

**Comment by CRITAWAKETS** [comment]
fairly certain the camera bug spawn in space ruins is made to be a source of illegal tech using an item that's basically just trash, so it's probably best to find something else to fill in that role. as for balancing, space ruins already can have alien tools so illegal tech's honestly not that concerning in comparison

**Comment by JohnFulpWillard** [comment]
Actually I decided against having these in maps at all since they are infinitely clonable

**Comment by JohnFulpWillard** [comment]
> fairly certain the camera bug spawn in space ruins is made to be a source of illegal tech using an item that's basically just trash, so it's probably best to find something else to fill in that role.

Not sure if the intent was for it to be used as illegal tech or was meant to be used for watching cameras while away- either way what alternatives do we have? Or maybe we can just add some useless space-only item that grants illegal tech.

**Comment by Vincent983** [comment]
could replace the spots that camera bugs were with syndicate toolboxes perhaps? I believe they give syndicate tech

**Comment by JohnFulpWillard** [comment]
Currently only 2 ruins have camera bugs in them, and syndicate toolboxes don't grant illegal tech.
maybe super pointy tape lol

**Comment by OrionTheFox** [comment]
One of the most underused but fun parts of the Camera Bug was that you could mark somebody to be "tracked", and while tracked it would give the name of whatever camera they're in view of - is that going to be carried over to this one?

**Comment by JohnFulpWillard** [comment]
> One of the most underused but fun parts of the Camera Bug was that you could mark somebody to be "tracked", and while tracked it would give the name of whatever camera they're in view of - is that going to be carried over to this one?

**Comment by carlarctg** [comment]
> Actually I decided against having these in maps at all since they are infinitely clonable

Just make a subtype that isn't...? And even then this isn't even a big deal. You are vastly overestimating people, nobody's going space ruin hunting and coming back to mass spread camera apps - hardly anyone uses pda apps to begin with. Feels like you're removing a neat thing from space (handheld camera monitor) for the sake of some invisible, ineffable Balance.

**Comment by SimplySmiley123** [comment]
> could replace the spots that camera bugs were with syndicate toolboxes perhaps? I believe they give syndicate tech. As for the intent, I think it was to be used as illegal tech considering how many ruins have them. (and the fact that they were broken for a few years)

They don't, tried it several times before but it's apparently not an item good enough for illegal tech.

**Comment by Vincent983** [comment]
so THAT explains why there's a syndicate toolbox on tram roundstart

**Comment by nukularpower** [comment]
Should not be added without a replacement for the illegal tech imo.   Syndi IDs work though, and there are a few of those in space, so not the end of the world.  If they work for illegals, maybe replace with EMP flashlights as a semi-counter to newly flashproof borgs?

BUT, camera bugs are pretty underrated - I actually like using them.  I am not sure many people know but the range of telekinesis is  higher than what you can see.   If you stand relatively close and use a camera bug you can TK things from off screen.   You can also use it to do remote upgrades with a bluespace RPED, which is nice.  Hopefully that stuff works with the new thing too.

Also yes tracking please :D

**Comment by newfren** [comment]
> Currently only 2 ruins have camera bugs in them, and syndicate toolboxes don't grant illegal tech. maybe super pointy tape lol

there's also a loot spawner for space ruins that places them or raw cores or old tech disks

**Comment by DrDiasyl** [comment]
The SecureVision application is bugged and doesn't worked. As I think this will be kinda based on it, will it fix that bug?

**Comment by JohnFulpWillard** [comment]
https://github.com/tgstation/tgstation/assets/53777086/a386536a-fe2a-4737-beea-0d69829b6acd

**Comment by Exester509** [comment]
> Currently only 2 ruins have camera bugs in them, and syndicate toolboxes don't grant illegal tech. maybe super pointy tape lol

Super pointy tape seems like a funny idea that still fills that niche

**Comment by OrionTheFox** [comment]
> 8mb.video-8ri-VGhM8z6B.mp4

Holy moly...

**Pull request merged**